tap_fixerio.py

```diff
@@ -45,21 +44,24 @@ def request(url, params):
     response = requests.get(url=url, params=params)
     response.raise_for_status()
     return response
-    
-def do_sync(base, start_date):
+
+def do_sync(base, start_date, access_key):
     logger.info('Replicating exchange rate data from fixer.io starting from {}'.format(start_date))
     singer.write_schema('exchange_rate', schema, 'date')
 
     state = {'start_date': start_date}
     next_date = start_date
-    
+
     try:
         while True:
-            response = request(base_url + next_date, {'base': base})
+            response = request(base_url + next_date, {'base': base, 'access_key': access_key})
             payload = response.json()
+            print(payload)
 
             if datetime.strptime(next_date, DATE_FORMAT) > datetime.utcnow():
                 break
+            elif payload.get('error'):
+                break
```

Comment:
Should the be a `raise` or at least a `logger.error` or something?

Reply:
Ah I see; I was following the convention of line 61, but line 61 is just quitting because it's done, not erring.

So this would be a good place regarding the access key hygiene. This is catching things like: API key invalid, API limit hit. So yeah, it should be logged and the other print removed. (line 59).

Reply:
I agree that "access key hygiene" should be maintained 👍 

If we encounter things like invalid API keys or hitting an API limit, then I think the `raise` is better choice here. Maybe something like:
```
elif payload.get('error'):
    raise APIError(payload['error'])
```

Reply:
Should I raise instead of log?

Reply:
I think so. If we just `log`, then I think the tap can still run. 

I would rather it fail if it hits an error.